pkg/controller/infrastructure/actuator_reconcile.go

```diff
@@ -121,9 +120,9 @@ func (a *actuator) maintainFirewallDeployment(ctx context.Context, logger logr.L
 	// a controller has no possibility to find out by itself if a reconciliation was triggered from the maintenance controller
 	// so it cannot be put to the worker controller.
 
-	spew.Dump(cluster)
-
 	if !gardener.EffectiveShootMaintenanceTimeWindow(cluster.Shoot).Contains(time.Now()) {
+		// note that this prevents updating the firewall image even when annotating the shoot explicitly with the maintainenance annotation
+		// if a user wants to update the firewall immediately he needs to specify the new firewall image in the spec
```

Comment:
This might not work in all cases, e.g. if the image is already at the latest version. Therefore i propose to skip this check if the reconcile annotation is present. WDYT ?

Reply:
Unfortunately the controller never sees a reconcile or maintenance annotation in these resources. If we remove this check it will update the firewall potentially outside the maintenance time window when changing something in the infrastructure config (e.g. when adding an egress rule or something like that).

So if a user wants like to immediately update outside the maintenance time window with a shorthand image, he needs to change to a specific firewall image instead.